src/lib.rs

```diff
@@ -56,9 +56,47 @@ impl Cli {
 
         let delete_file = run::generate_delete_file(tmp_dir.path())?;
 
-        run::start_shell(config, tmp_dir.path())?;
+        let mut shell = match run::start_shell(config, tmp_dir.path()) {
+            Ok(shell) => shell,
+            Err(err) => {
+                run::clean_up(
+                    delete_file,
+                    tmp_dir,
+                    self.worktree_branch.clone(),
+                    Vec::new(),
+                )?;
+                bail!("{}", err);
+            }
+        };
+
+        let subprocesses = run::start_subprocesses(config, tmp_dir.path());
 
-        run::clean_up(delete_file, tmp_dir, self.worktree_branch.clone())?;
+        #[cfg(windows)]
+        if config.editor.is_some() {
+            unsafe {
+                cargo_temp_bindings::Windows::Win32::SystemServices::FreeConsole();
+            }
+        }
+
+        match shell.wait() {
+            Ok(_) => {
+                run::clean_up(
+                    delete_file,
+                    tmp_dir,
+                    self.worktree_branch.clone(),
+                    subprocesses,
+                )?;
+            }
+            Err(err) => {
+                run::clean_up(
+                    delete_file,
+                    tmp_dir,
+                    self.worktree_branch.clone(),
+                    subprocesses,
+                )?;
+                bail!("cannot wait shell process: {}", err);
+            }
+        }
```

Comment:
You can run `shell.wait()` outside the match and store it in a `res` variable. Then you run clean_up (so you have no duplicated code), then you do something with `res` :bulb: